File: jeorg-kotlin-coroutines/coroutines-crums-group-1/src/main/java/org/jesperancinha/ktd/java/ValidationUsersEmail.java
package org.jesperancinha.ktd.java;

import static jakarta.validation.Validation.buildDefaultValidatorFactory;

    public class ValidationUsersEmail {
        public static void main(String[] args) {
            try(var factory = buildDefaultValidatorFactory()) {
                var validator = factory.getValidator();
                var validate1 = validator.validate(new User("bad"));
                System.out.println(validate1);
                var validate2 = validator.validate(new User("user@someemail.com"));
                System.out.println(validate2);
            }
        }
    }

File: jeorg-kotlin-coroutines/coroutines-crums-group-1/src/main/kotlin/org/jesperancinha/ktd/ValidationUsersEmail.kt
package org.jesperancinha.ktd

import jakarta.validation.Validation
import jakarta.validation.constraints.Email


data class User (
    @field:Email(message = "Please insert a valid email")
    val email:String
)

class ValidationUsersEmail {
  companion object {
      @JvmStatic
      fun main(args: Array<String> = emptyArray()) {
          val factory = Validation.buildDefaultValidatorFactory()
          val validator = factory.validator
          val validate1 = validator.validate(User("bad"))
          println(validate1)
          val validate2 = validator.validate(User("user@someemail.com"))
          println(validate2)
      }
  }
}